index.js

```diff
@@ -9,11 +9,19 @@ export default class BananaSlug {
   /**
    * Create a new slug class.
    */
-  constructor () {
+  constructor (previousOccurrences) {
     /** @type {Record<string, number>} */
     // eslint-disable-next-line no-unused-expressions
-    this.occurrences
-
+    if(previousOccurrences){
+      this.occurrences = previousOccurrences.reduce((obj, slug) => {
+        const strippedSlug = slug.replace(/-\d+$/, '');
+        obj[strippedSlug] = (obj[strippedSlug] || 0) + 1;
+        return obj;
+      }, {});
+    }
```

Comment:
Hey!

- how do you want to save this stuff to a database?
- the `strippedSlug` part comes with a bug: what if someone writes a heading `## Heading 1`?
- I think the `this.reset` below overwrites all your changes
- docs and tests!
- lastly, the previous points can be addressed first, but there’s also a security problem here: a slug of `constructor` will be injected into the object, and cause problems!

---

I’d first like to hear more about your use case.
Perhaps we can come up with a different API. Maybe like a `slugger.toJSON()` function which yields the map, and a `Slugger.fromJSON(map)` to revive it?